src/Shared/TempFileUtilities.cs

```diff
@@ -36,7 +38,18 @@ internal static void ClearTempFileDirectory()
         // For all native calls, directly check their return values to prevent bad actors from getting in between checking if a directory exists and returning it.
         private static string CreateFolderUnderTemp()
         {
-            string basePath = Path.Combine(Path.GetTempPath(), $"MSBuildTemp{Environment.UserName}");
+            string msbuildTempFolder;
+            // On windows Username with Unicode chars can give issues, so we dont append username to the temp folder name.
+            if (NativeMethodsShared.IsWindows)
+            {
+                msbuildTempFolder = msbuildTempFolderPrefix;
+            }
+            else
+            {
+                msbuildTempFolder = msbuildTempFolderPrefix + Environment.UserName;
+            }
```

Comment:
nit:
```suggestion
            // On windows Username with Unicode chars can give issues, so we dont append username to the temp folder name.
            string msbuildTempFolder = NativeMethodsShared.IsWindows ?
                msbuildTempFolderPrefix :
                msbuildTempFolderPrefix + Environment.UserName;
```

I'm a big fan of ternary operators, especially at initialization time like this, but I recognize not everyone agrees 🙂

Reply:
Depends on the situation, but here it makes perfect sense :) thanks!